### PR TITLE
Refactor LocalDataStore

### PR DESCRIPTION
**Pull request by gino-m**
- [x] Move mutation coordination methods into `MutationRepository`
- [x] Inject local data stores directly instead of via `LocalDataStore`
- [x] Delete `LocalDataStore`
- [x] Preserve client timestamps passed through `LocationOfInterestMutation`

Towards #1491. See also #1559, #1561, #1562.

@shobhitagarwal1612 and/or @scolsen PTAL?

### CODE REVIEW

File: ground/src/main/java/com/google/android/ground/model/locationofinterest/LocationOfInterest.kt
Comment (shobhitagarwal1612) [inline ground/src/main/java/com/google/android/ground/model/locationofinterest/LocationOfInterest.kt:69]: Do we update `lastModified` before creating a mutation? I thought the purpose of using `Date()` was to ensure that the timestamp always matches when the mutation is created. Can you please check?

**Review by shobhitagarwal1612 — APPROVED** [review]
Approving the PR to unblock but please address the above comment before merging. Thanks!

File: ground/src/main/java/com/google/android/ground/model/locationofinterest/LocationOfInterest.kt
Comment (gino-m) [reply ground/src/main/java/com/google/android/ground/model/locationofinterest/LocationOfInterest.kt:69]: I don't think this belongs in the model.. It's business logic that needs to be managed at the appropriate layer in the code, likely in the repository. This codepath is currently unused, but should be set the right thing once we allow adding LOIs in the data collection flow.

**Comment by gino-m** [comment]
> Approving the PR to unblock but please address the above comment before merging. Thanks!

Thanks for the review, Shobhit!

**Pull request merged**
